extension/wrapper.cpp

```diff
@@ -116,6 +116,45 @@ Satrec_twoline2rv(PyTypeObject *cls, PyObject *args)
     return (PyObject*) self;
 }
 
+static PyObject *
+Satrec_sgp4init(PyObject *self, PyObject *args)
+{
+    long int satnum;
+    double jdSGP4epoch, bstar, ndot, nddot;
+    double ecco, argpo, inclo, mo, no_kozai, nodeo;
+
+    if (!PyArg_ParseTuple(args, "ldddddddddd:sgp4init", &satnum, &jdSGP4epoch, &bstar, &ndot, &nddot,
+                          &ecco, &argpo, &inclo, &mo, &no_kozai, &nodeo))
+        return NULL;
+
+    elsetrec &satrec = ((SatrecObject*) self)->satrec;
+
+    SGP4Funcs::sgp4init(wgs72, 'i', satnum, jdSGP4epoch, bstar, ndot, nddot,
+                        ecco, argpo, inclo, mo, no_kozai, nodeo, satrec);
+
+    /* Return true as sgp4init does, satrec.error contains any error codes */
+
+    /* Present state: sgp4init runs fine, but function segmentation faults
+    at the return statement */
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+static PyObject *
+Satrec_sgp4_tsince(PyObject *self, PyObject *args)
+{
+    double tsince, r[3], v[3];
+
+    if (!PyArg_ParseTuple(args, "d:sgp4_tsince", &tsince))
+        return NULL;
+    elsetrec &satrec = ((SatrecObject*) self)->satrec;
+    SGP4Funcs::sgp4(satrec, tsince, r, v);
+    if (satrec.error && satrec.error < 6)
+        r[0] = r[1] = r[2] = v[0] = v[1] = v[2] = NAN;
+    return Py_BuildValue("i(f,f,f)(f,f,f)", satrec.error,
```

Comment:
Fixed. This was part of exploring returning something other than a tuple.

I'd still love for this to return a C-array (or python array.array) without touching numpy, but have not figured that out yet.  For these 3d vectors, raw routines with python array.array are significantly (3-8x) faster than numpy routines.

Reply:
Can you call the array-based method with a 1-element time array? That should give you NumPy arrays back without the expense of intermediate Python representations. Let me know if you can't get that to work, and I can suggest sample code! (And will them update the docs to make the approach clearer.)

Reply:
I was more speaking of the r,v vectors as r[3] and v[3] C-arrays

Reply:
Tried calling the array-based method with a 1-element time array to get a numpy array "for free", but it comes at a penalty of being about 3x slower.  The crossover between multiple calls of the single method vs. the array method appears to be around ~5 elements in my environment.  Given that I'm almost always dealing with 3 or more time-elements, it might be worth favoring the array method anyway.  I'll explore this a little more thoroughly with a closer evaluation of the performance hit of the totality of the array-setup overhead.

Reply:
FYI - Finished implementing an all array (_sgp4) implementation in my own code.  In the course of doing that, also explored the openmp "parallel for" options.  I tried various optimizations of your default code, but generally saw that it was 2.5x slower for small numbers (10) and 60% slower for (100) elements on an 18-core machine with openmp from `brew install llvm libomp`.  At ten entries, the "no openmp" version is about 20% faster, but don't have test data to compare at 100 element.